### PR TITLE
Issue 295: Context strategy Checker patch filter shouldn't contain hardcoded checks

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue 295: Context strategy Checker patch filter shouldn't contain hardcoded checks

### CODE REVIEW

**Comment by HuGanghui** [comment]
in my local, all tests passed, and in CI fails, wired.

**Comment by baratali** [comment]
Test fails because folder name is `neverSuppressed`, but in test it is in lowercase `neversuppressed`

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchfilter/FileLength/neversuppressed/case1/defaultContextConfig.xml [hunk 1]
```diff
@@ -8,6 +8,6 @@
   </module>
   <module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchFilter">
     <property name="file" value="${tp}/defaultContext.patch" />
-    <property name="strategy" value="patchedline" />
```
Comment (baratali) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchfilter/FileLength/neversuppressed/case1/defaultContextConfig.xml:11]: why do you change strategy for these checks?
I think strategy should stay and you just add neverSuppressedChecks property.

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java [hunk 2]
```diff
@@ -126,25 +125,6 @@ private boolean isLineMatching(AuditEvent event) {
         return result;
     }
 
-    /**
```
Comment (baratali) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java:129]: please add `#` in commit message
`Issue 295: Context strategy Checker patch filter shouldn't contain hardcoded checks`
-->
`Issue #295: Context strategy Checker patch filter shouldn't contain hardcoded checks`

File: src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java [hunk 3]
```diff
@@ -48,11 +48,11 @@ public void testNeverSuppressedChecks() throws Exception {
 
     @Test
     public void testFileLength() throws Exception {
-        testByConfig("FileLength/newline/defaultContextConfig.xml");
```
Comment (baratali) [inline src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java:51]: folder names also should stay the same

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchfilter/FileLength/neversuppressed/case1/defaultContextConfig.xml (hunk 1 above)
Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchfilter/FileLength/neversuppressed/case1/defaultContextConfig.xml]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java (hunk 2 above)
Comment (HuGanghui) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java:129]: done

File: src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java (hunk 3 above)
Comment (HuGanghui) [reply src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java:51]: sorry, I am not clear, current version is not same?

**Comment by HuGanghui** [comment]
@baratali 
>Test fails because folder name is neverSuppressed, but in test it is in lowercase neversuppressed

oh, yes, thanks for your help

File: src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java (hunk 3 above)
Comment (baratali) [reply src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java:51]: I mean do not remove strategy name from the folder name. It should be consistent with other tests - `<Check>/<strategy>`. I would not rename any of these resources.
If you still want to group them under neversuppressed folder, then do this under strategy folder.
Example:
```
FileLength/newline/neversuppressed/defaultContextConfig.xml
FileLength/patchedline/neversuppressed/defaultContextConfig.xml
```

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchfilter/Header/neversuppressed/case2/defaultContextConfig.xml [hunk 4]
```diff
@@ -10,6 +10,6 @@
 
   <module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchFilter">
     <property name="file" value="${tp}/defaultContext.patch" />
-    <property name="strategy" value="patchedline" />
```
Comment (baratali) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchfilter/Header/neversuppressed/case2/defaultContextConfig.xml:13]: do not remove strategy

File: src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java (hunk 3 above)
Comment (HuGanghui) [reply src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java]: done

**Review by baratali — APPROVED** [review]
looks good

**Pull request merged**